genmod/score_variants/score_variant.py

```diff
@@ -80,36 +96,38 @@ def get_category_score(variant, category, config_parser, csq_format=None):
             category_score = sum(category_scores)
             logger.debug("Sum of scores is {0}".format(
                 category_score))
-        
+
     else:
         logger.debug("No scores found for category {0}".format(category))
-    
-    return category_score
-    
 
-def score_variant(variant, config_parser, csq_format=None):
-    """Score a variant line
-    
-        Args:
-            variant (dict): A variant dictionary
-            config_parser (ConfigParser): A config parser object with score functions
-        Returns:
-            rank_score (float): The rank score for this variant
-    """
-    
-    rank_score = 0
-    
-    for category in config_parser.categories:
-        logger.debug("Checking scores for category {0}".format(category))
-        category_score = get_category_score(
-            variant = variant,
-            category = category,
-            config_parser = config_parser,
-            csq_format = csq_format
-        )
+    if not (category_score_min <= category_score <= category_score_max):
+        raise ValueError('Category score outside expected category score range',
+                          category_score_min, category_score, category_score_max)
+    return category_score, category_score_min, category_score_max
 
-        logger.debug("Adding category score {0} to rank_score".format(category_score))
-        rank_score += category_score
-        logger.debug("Updating rank score to {0}".format(rank_score))
 
-    return rank_score
+def as_normalized_max_min(score: float,
+                          min_score_value: float,
+                          max_score_value: float) -> float:
+    """
+    Performs max-min normalization of score to range (0, 1).
+    Args:
+        score: Rank score
+        min_score_value: Lower bound on score
+        max_score_value: Upper bound on score
+    Returns:
+        Score in range (0, 1)
+    """
+    if not isinstance(score, float):
+        raise TypeError('Bad score type', score)
+    if not isinstance(min_score_value, float):
+        raise TypeError('Bad min_score_value value', min_score_value)
+    if not isinstance(max_score_value, float):
+        raise TypeError('Bad max_score_value value', max_score_value)
```

Comment:
Make a dict of {"variable_name": variable}

Loop over key and value and check isinstance. Keep it DRY and easier to extend in the future without having to change logic